src/main/java/meteordevelopment/meteorclient/systems/modules/render/Tracers.java

```diff
@@ -192,6 +262,141 @@ else if (entity instanceof PlayerEntity) {
         }
     }
 
+    @EventHandler
+    public void onRender2D(Render2DEvent event) {
+        if (mc.options.hudHidden || style.get() != TracerStyle.Offscreen) return;
+        count = 0;
+
+        for (Entity entity : mc.world.getEntities()) {
+            if (!PlayerUtils.isWithin(entity, maxDist.get()) || (!Modules.get().isActive(Freecam.class) && entity == mc.player) || !entities.get().getBoolean(entity.getType()) || (ignoreFriends.get() && entity instanceof PlayerEntity && Friends.get().isFriend((PlayerEntity) entity)) || (!showInvis.get() && entity.isInvisible()) | !EntityUtils.isInRenderDistance(entity)) continue;
+
+            Color color;
+
+            if (distance.get()) {
+                if (friendOverride.get() && entity instanceof PlayerEntity && Friends.get().isFriend((PlayerEntity) entity)) {
+                    color = Config.get().friendColor.get();
+                }
+                else color = EntityUtils.getColorFromDistance(entity);
+            }
+            else if (entity instanceof PlayerEntity) {
+                color = PlayerUtils.getPlayerColor(((PlayerEntity) entity), playersColor.get());
+            }
+            else {
+                color = switch (entity.getType().getSpawnGroup()) {
+                    case CREATURE -> animalsColor.get();
+                    case WATER_AMBIENT, WATER_CREATURE, UNDERGROUND_WATER_CREATURE, AXOLOTLS -> waterAnimalsColor.get();
+                    case MONSTER -> monstersColor.get();
+                    case AMBIENT -> ambientColor.get();
+                    default -> miscColor.get();
+                };
+            }
+
+            Color finalColor = new Color(color.r, color.g, color.b, color.a);
```

Comment:
You can just modify the existing color object, no need to create a new one

Reply:
unfortunately doesn't seem to be the case as color becomes a reference and editing it changes the setting itself.